crates/ty_python_semantic/src/types.rs

```diff
@@ -385,7 +406,7 @@ pub(crate) use todo_type;
 /// # Ordering
 /// Ordering is based on the property instance's salsa-assigned id and not on its values.
 /// The id may change between runs, or when the property instance was garbage collected and recreated.
-#[salsa::interned(debug, heap_size=ruff_memory_usage::heap_size)]
+#[salsa::interned(persist, revisions = usize::MAX, debug, heap_size=ruff_memory_usage::heap_size)]
```

Comment:
I disabled garbage collection for all interned values because it significantly reduces the amount of data we need to persist, but I'm not sure if that change should be made in this PR.

Reply:
I think this should be a separate PR and requires some reasoning why it's fine to do this